tests/integration-tests/tests/capacity_reservations/test_capacity_blocks.py

```diff
@@ -0,0 +1,110 @@
+from utils import retrieve_cfn_resources
+
+
+@pytest.mark.usefixtures("os", "scheduler")
+def test_capacity_blocks(
+    region, pcluster_config_reader, odcr_stack, clusters_factory, test_datadir, scheduler_commands_factory
+):
+    """
+    Test that clustermgtd and the system works as expected when using Capacity Blocks.
+
+    The Capacity Blocks behaviour is totally simulated, by using an ODCR and some mocking and overrides.
+
+    Fleet-config and describe-capacity-reservations output will be mocked in the following way:
+    - static nodes initially will be created because part of an ODCR
+    - fleet-config.json will be modified to consider the Capacity Reservation a CB, rather than an ODCR
+    - the describe_capacity_reservations will be mocked to return a pending/active CB to simulate state transition
+    The test will verify that Slurm reservations are created/deleted accordingly and job in pending will start.
+    """
+    # We are using an ODCR to simulate a CB
+    odcr_resources = retrieve_cfn_resources(odcr_stack.name, region)
+    capacity_block_id = odcr_resources["integTestsTargetOdcr"]
+    cluster_config = pcluster_config_reader(capacity_reservation_id=capacity_block_id)
+    cluster = clusters_factory(cluster_config)
+    reservation_name = f"pcluster-{capacity_block_id}"
+
+    # override describe-capacity-reservations output
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    remote_command_executor.run_remote_script(
+        script_file=str(test_datadir / "override_capacity_blocks.sh"), args=[capacity_block_id]
+    )
+    logging.info("Restart clustermgtd to load the overrides module")
+    remote_command_executor.run_remote_command(command="sudo systemctl restart supervisord")
+    # wait 1 minute for clustermgtd execution
+    time.sleep(60)
+    _check_slurm_reservation_existence(remote_command_executor, capacity_block_id, reservation_name)
+
+    # submit a job that will remain in pending because CB is not active
+    logging.info("Submitted a job in a pending CB")
+    scheduler_commands = scheduler_commands_factory(remote_command_executor)
+    result = scheduler_commands.submit_command("sleep 1")
+    job_id = scheduler_commands.assert_job_submitted(result.stdout)
+
+    # Simulate active CB and check slurm reservations are removed
+    _modify_capacity_block_override_status(remote_command_executor, previous_status="pending", new_status="active")
+    # Verify slurm reservation has been removed
+    logging.info("Checking slurm reservation for CB %s has been removed", capacity_block_id)
+    result = remote_command_executor.run_remote_command(
+        command=f"scontrol show ReservationName={reservation_name}", raise_on_error=False
+    )
+    assert_that(result.stdout).contains(f"Reservation {reservation_name} not found")
+
+    # Verify job is completed
+    logging.info(f"Job {job_id} that was pending in the inactive CB, is now completed")
```

Comment:
move this log after the wait_job_completed

Reply:
moved, thanks!